dhis-2/dhis-services/dhis-service-expression-parser/src/test/java/org/hisp/dhis/expressionparser/ExpressionParserServiceTest.java

```diff
@@ -0,0 +1,1148 @@
+        dataSetA.addOrganisationUnit( orgUnitH );
+        dataSetA.addOrganisationUnit( orgUnitI );
+
+        dataSetB.addOrganisationUnit( orgUnitF );
+        dataSetB.addOrganisationUnit( orgUnitG );
+        dataSetB.addOrganisationUnit( orgUnitI );
+
+        dataSetService.addDataSet( dataSetA );
+        dataSetService.addDataSet( dataSetB );
+
+        reportingRateA = new ReportingRate( dataSetA, REPORTING_RATE );
+        reportingRateB = new ReportingRate( dataSetA, REPORTING_RATE_ON_TIME );
+        reportingRateC = new ReportingRate( dataSetA, ACTUAL_REPORTS );
+        reportingRateD = new ReportingRate( dataSetA, ACTUAL_REPORTS_ON_TIME );
+        reportingRateE = new ReportingRate( dataSetA, EXPECTED_REPORTS );
+        reportingRateF = new ReportingRate( dataSetB );
+
+        reportingRateA.setUid( "reportRateA" );
+        reportingRateB.setUid( "reportRateB" );
+        reportingRateC.setUid( "reportRateC" );
+        reportingRateD.setUid( "reportRateD" );
+        reportingRateE.setUid( "reportRateE" );
+        reportingRateF.setUid( "reportRateF" );
+
+        indicatorTypeA = new IndicatorType( "A", 100, false );
+
+        Constant constantA = new Constant( "One half", 0.5 );
+        Constant constantB = new Constant( "One quarter", 0.25 );
+
+        constantA.setUid( "xxxxxxxxx05" );
+        constantB.setUid( "xxxxxxxx025" );
+
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+
+        constantMap = constantService.getConstantMap();
+
+        valueMap = entries (
+            entry( dataElementA, 3.0 ),
+            entry( dataElementB, 13.0 ),
+
+            entry( dataElementOperandA, 5.0 ),
+            entry( dataElementOperandB, 15.0 ),
+            entry( dataElementOperandC, 7.0 ),
+            entry( dataElementOperandD, 17.0 ),
+            entry( dataElementOperandE, 9.0 ),
+            entry( dataElementOperandF, 19.0 ),
+
+            entry( programDataElementA, 101.0 ),
+            entry( programDataElementB, 102.0 ),
+
+            entry( programAttributeA, 201.0 ),
+            entry( programAttributeB, 202.0 ),
+
+            entry( programIndicatorA, 301.0 ),
+            entry( programIndicatorB, 302.0 ),
+
+            entry( reportingRateA, 401.0 ),
+            entry( reportingRateB, 402.0 ),
+            entry( reportingRateC, 403.0 ),
+            entry( reportingRateD, 404.0 ),
+            entry( reportingRateE, 405.0 ),
+            entry( reportingRateF, 406.0 )
+        );
+    }
+
+    // -------------------------------------------------------------------------
+    // Supportive methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Adds a list of entry lists to the value map.
+     *
+     * @param entries the lists of entries to add.
+     * @return the value map.
+     */
+    private <K, V> Map<K, V> entries( Map.Entry<K, V>... entries )
```

Comment:
This causes a "Type safety: Potential heap pollution via varargs parameter entries" warning in IDE due to lack of type checks.

Replace with the use of Guava Map Builder: 

Map<K, V> map = ImmutableMap.<K, V>builder()
  .put( a, b )
  .build();

https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html

Reply:
Done.